src/CopyrighFixer/CopyrighFixer/signer.cpp

```diff
@@ -7,13 +7,55 @@
 
 #include "signer.h"
 #include "CopyrighFixer/ifilemanager.h"
+#include <QDir>
 
 namespace CopyrighFixer {
 Signer::Signer() {
 
 }
 
-bool Signer::checkSign(const Config &objConf) {
+bool Signer::checkSign(const Config &objConf) {    
+
+    Config currConfig = objConf;
+
+    QDir currentFolder(currConfig.getSrcDir());
+    currentFolder.setFilter(QDir::Dirs | QDir::Files);
+
+    QFileInfoList folderItems(currentFolder.entryInfoList());
+
+    for (auto i_file: folderItems) {
+
+        if (i_file.fileName() == "." || i_file.fileName() == "..") {
+            continue;
+        }
+
+        if (!i_file.isDir()) {
+
+            IFileManager *currFM = searchFileByExt(i_file.suffix());
+            Signature objSing = currConfig.getSignVal();
+
+            if (currFM == nullptr) {
+                return false;
+            }
+
+            bool valRead = currFM->read(i_file.filePath(), objSing);
+            if (!valRead) {
+                return false;
+            }
+
+            currConfig.setSingValue(objSing);
```

Comment:
Your method absolutely wrong because you read a signature from file and replace without processing a files signature to  an user signature from config and save new signature.

I think you need to check signature owners. if signature of the file have a difference of the user signature then you need to show warning about and do not update old signature. else you need update just owner and date.